app/src/main/java/me/guillaumin/android/osmtracker/db/DataHelper.java

```diff
@@ -65,6 +68,9 @@
 	 */
 	private Context context;
 
+	/** Statistics for all existing tracks */
+	private static TreeMap<Long, TrackStatistics> tracksStatistics = new TreeMap<Long, TrackStatistics> ();
```

Comment:
I guess there's also the performance aspect to consider as you said initially. If it turns out computing these values is too slow any way on big tracks we'll have to store it in the DB and that'll change the approach completely. Perhaps it's worth trying to run it with a big track with thousands of track points and see how the list view performs?